lightning/src/ln/channelmanager.rs

```diff
@@ -1839,9 +1842,19 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		let mut channel_state = Some(self.channel_state.lock().unwrap());
 		let removed_source = channel_state.as_mut().unwrap().claimable_htlcs.remove(&(payment_hash, *payment_secret));
 		if let Some(mut sources) = removed_source {
+			assert!(!sources.is_empty());
+			let passes_value = if let &Some(ref data) = &sources[0].payment_data {
```

Comment:
Heh, val commented the same. The next commit changes it to valid_mpp_amount instead.

As for the second part of your comment.......ugh. We can't get partial claims...except that we were releasing the channel_state lock between each claim so theoretically we could (and also because we don't have support for claiming funds from a since-closed channel, though we will eventually gain support for that). I added a new commit to address this issue.

Reply:
Hmm, no, it wasnt meant to be - once we're sending over multiple paths we have to make sure we check the value before we start, instead of where it is for non-MPP payments. I can pull it out if you really want, but its logically required for sending over multiple paths